src/core/attribution/markovChain.js

```diff
@@ -118,26 +132,27 @@ function* findStationaryDistributionGenerator(
     });
     return maxDelta;
   }
-  let iteration = 0;
+  let nIterations = 0;
+  let convergenceDelta = NaN;
   while (true) {
-    if (iteration >= options.maxIterations) {
+    if (nIterations >= options.maxIterations) {
       if (options.verbose) {
-        console.log(`[${iteration}] FAILED to converge`);
+        console.log(`[${nIterations}] FAILED to converge`);
       }
-      return pi;
+      return {distribution, nIterations, convergenceDelta};
     }
-    iteration++;
-    sparseMarkovChainActionInto(chain, pi, scratch);
-    const delta = computeDelta(pi, scratch);
-    [scratch, pi] = [pi, scratch];
+    nIterations++;
+    sparseMarkovChainActionInto(chain, distribution, scratch);
+    convergenceDelta = computeDelta(distribution, scratch);
+    [scratch, distribution] = [distribution, scratch];
```

Comment:
I’m not going to request that this be reverted, but IMHO `pi` is much
clearer than `distribution`. _Both_ `scratch` and `pi` are
distributions, but one is scratch space and the other is our
current-best approximation of the stationary distribution.

Reply:
Ack.